src/wasm.cc

```diff
@@ -526,6 +570,7 @@ std::shared_ptr<WasmHandleBase> createWasm(const std::string &vm_key, const std:
 std::shared_ptr<WasmHandleBase> getThreadLocalWasm(std::string_view vm_key) {
   auto it = local_wasms.find(std::string(vm_key));
   if (it == local_wasms.end()) {
+    removeStaleLocalCacheEntries(local_wasms, local_wasms_keys);
```

Comment:
Suggest reordering this function to be:

```
std::shared_ptr<WasmHandleBase> getThreadLocalWasm(std::string_view vm_key) {
  auto it = local_wasms.find(std::string(vm_key));
  if (it != local_wasms.end()) {
    auto wasm = it->second.lock();
    if (wasm) {
      return wasm;
    }
  }
  removeStaleLocalCacheEntries(local_wasms, local_wasms_keys);
  return nullptr;
}
```

This would ensure that `removeStaleLocalCacheEntries` is called in all miss cases (currently it doesn't get called if `it != local_wasms.end` but `wasm == nullptr`), and also makes the structure similar to `getOrCreateThreadLocalWasm` and `getOrCreateThreadLocalPlugin`.

Reply:
Right, good point, done.